databox/api/src/Integration/Happyscribe/HappyscribeAction.php

```diff
@@ -0,0 +1,175 @@
+                ->getAttributeDefinitionBySlug($asset->getWorkspaceId(), $config['sourceLanguageAttribute'])
+                    ?? throw new \InvalidArgumentException(sprintf('Attribute definition slug "%s" not found in workspace "%s"', $config['sourceLanguageAttribute'], $asset->getWorkspaceId()));
+
+        $sourceLanguage = $attributeIndex->getAttribute($srcLanguageAttrDef->getId(), AttributeInterface::NO_LOCALE)?->getValue() ?? $config['defaultSourceLanguage'];
+        $sourceLanguage = trim($sourceLanguage);
+
+        if (!in_array(strlen($sourceLanguage), [2, 5])) {
+            $t = explode('-', $sourceLanguage);
+            $sourceLanguage = $t[0];
+
+            if (2 != strlen($sourceLanguage)) {
+                throw new \InvalidArgumentException('Source language code must be a 2-letter or 4-letter code, eg: en, fr-FR, ...');
+            }
+        }
+
+        try {
+            $responseTranscription = $this->happyscribeClient->request('POST', 'https://www.happyscribe.com/api/v1/transcriptions', [
+                'headers' => [
+                    'Authorization' => 'Bearer '.$this->happyscribeToken,
+                ],
+                'json' => [
+                    'transcription' => [
+                        'name' => $asset->getTitle(),
+                        'is_subtitle' => true,
+                        'language' => $sourceLanguage,
+                        'organization_id' => $organizationId,
+                        'tmp_url' => $tmpUrl,
+                    ],
+                ],
+            ]);
+
+        } catch (\Exception $e) {
+            throw new \RuntimeException('Error when creating transcript : '.$e->getMessage());
+        }
+
+        if (200 !== $responseTranscription->getStatusCode()) {
+            throw new \RuntimeException('error when creating transcript,response status : '.$responseTranscription->getStatusCode().' message : '.$responseTranscription->getContent(false));
+        }
+
+        $responseTranscriptionBody = $responseTranscription->toArray();
+        $transcriptionId = $responseTranscriptionBody['id'];
+
+        $this->happyscribeToken =
+        $this->extension = $config['transcriptFormat'];
+        $organizationId = $config['organizationId'];
+
+        $options['apiKey'] = $config['apiKey'];
+        $options['transcriptFormat'] = $config['transcriptFormat'];
+        $options['attribute'] = $config['attribute'];
+        $options['sourceLanguageAttribute'] = $config['sourceLanguageAttribute'];
+        $options['defaultSourceLanguage'] = $config['defaultSourceLanguage'];
+        $options['assetId'] = $asset->getId();
+        $options['workspaceId'] = $asset->getWorkspaceId();
+        $options['sourceLanguage'] = $sourceLanguage;
+        $options['attributeId'] = $attrDef->getId();
+        $options['isTranslatableAttribute'] = $attrDef->isTranslatable();
+        $options['isMultipleAttribute'] = $attrDef->isMultiple();
```

Comment:
Just pass the integration ID and the minimum information you need in this context